File: models/linked_clan.py
"""
Модель данных для привязанных кланов
"""
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class LinkedClan:
    """Привязанный клан пользователя"""
    id: Optional[int]
    telegram_id: int
    clan_tag: str
    clan_name: str
    slot_number: int  # номер слота (1-5)
    created_at: str
    
    def __init__(self, telegram_id: int, clan_tag: str, clan_name: str, 
                 slot_number: int, created_at: str = None, id: int = None):
        self.id = id
        self.telegram_id = telegram_id
        self.clan_tag = clan_tag
        self.clan_name = clan_name
        self.slot_number = slot_number
        self.created_at = created_at or datetime.now().isoformat()

File: internal/models/linked_clan.go
package models

import "time"

// LinkedClan represents a linked clan in the system
type LinkedClan struct {
	ID          int64     `json:"id"`
	TelegramID  int64     `json:"telegram_id"` // Alias for LinkedBy
	ClanTag     string    `json:"clan_tag"`
	ClanName    string    `json:"clan_name"`
	LinkedBy    int64     `json:"linked_by"` // telegram_id of user who linked
	LinkedAt    time.Time `json:"linked_at"`
	IsActive    bool      `json:"is_active"`
	Description *string   `json:"description,omitempty"`
	SlotNumber  int       `json:"slot_number"`
	CreatedAt   time.Time `json:"created_at"`
}

// NewLinkedClan creates a new LinkedClan instance
func NewLinkedClan(clanTag, clanName string, linkedBy int64) *LinkedClan {
	now := time.Now()
	return &LinkedClan{
		ID:         0, // Set by database
		TelegramID: linkedBy,
		ClanTag:    clanTag,
		ClanName:   clanName,
		LinkedBy:   linkedBy,
		LinkedAt:   now,
		IsActive:   true,
		SlotNumber: 0,
		CreatedAt:  now,
	}
}
